packages/core/upload/server/controllers/validation/admin/folder.js

```diff
@@ -15,7 +15,12 @@ const validateCreateFolderSchema = yup
       .min(1)
       .matches(NO_SLASH_REGEX, 'name cannot contain slashes')
       .matches(NO_SPACES_AROUND, 'name cannot start or end with a whitespace')
-      .required(),
+      .required()
+      .test('is-folder-unique', 'name already taken', async function(name) {
+        const { exists } = getService('folder');
+        const doesExist = await exists({ parent: this.parent.parent || null, name });
```

Comment:
I'm wondering if the `this.parent.parent` is a typo? Originally it was `this.parent` :)

Reply:
It is not :) It was `folder.parent`, since the function was moved a level down, I don't have the parent anymore but just the name. To retrieve the parent I have to get the value "above" and to get it it's `this.parent` and then my field `parent` => `this.parent.parent`